### PR TITLE
Refactor/373 refactor buy asset form

### PR DESCRIPTION
**Pull request by jvrastil**
- needs to be merged after Exchange form refactoring

### CODE REVIEW

File: lib/logic/bloc/asset_exchange_form/asset_exchange_form_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+    if (v == NumpadValue.decimal) return curr.add(null);
+    return curr;
+  }
+
+  Future<void> _onValueConverted(
+    ValueConverted event,
+    Emitter<ExchangeFormState> emit,
+  ) async {
+    // TODO - implement
+    // print('CONVERTING VALUE');
```
Comment (TomasPetrik) [inline lib/logic/bloc/asset_exchange_form/asset_exchange_form_bloc.dart:125]: Remove unused print() methods.
Even better use everytime printDebug()

File: lib/logic/bloc/asset_exchange_form/asset_exchange_form_bloc.dart [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    } catch (e) {
+      return [];
+    }
+  }
+
+  void _onValueSelected(
+    ValueSelected event,
+    Emitter<ExchangeFormState> emit,
+  ) {
+    // print('ON VALUE SELECTED');
```
Comment (TomasPetrik) [inline lib/logic/bloc/asset_exchange_form/asset_exchange_form_bloc.dart:202]: Remove commeted code.

File: lib/logic/bloc/asset_exchange_form/asset_exchange_form_bloc.dart [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+      source: s.source.copyWith(currencyType: ExchangeCurrencyType.crypto),
+      target: s.target.copyWith(currencyType: ExchangeCurrencyType.crypto),
+    ));
+  }
+
+  void _onValueSwitched(
+    ValueSwitched event,
+    Emitter<ExchangeFormState> emit,
+  ) {
+    // print('ON VALUE CHANGED');
```
Comment (TomasPetrik) [inline lib/logic/bloc/asset_exchange_form/asset_exchange_form_bloc.dart:217]: Same here.

File: lib/logic/bloc/asset_exchange_form/asset_exchange_form_bloc.dart [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+    final s = state;
+    if (s is! ExchangeConverted) return;
+
+    // print('Direction is: ${s.direction}');
```
Comment (TomasPetrik) [inline lib/logic/bloc/asset_exchange_form/asset_exchange_form_bloc.dart:221]: Remove commented code.

File: lib/view/authenticated_app.dart [hunk 5]
```diff
@@ -60,6 +60,7 @@ class AuthenticatedApp extends StatelessWidget {
           ),
         ),
         BlocProvider(create: (_) => TabBarCubit()),
+        // TODO - remove the cubit from here
         BlocProvider(
           create: (context) => AssetExchangeFormCubit(
```
Comment (TomasPetrik) [inline lib/view/authenticated_app.dart:65]: So now we have 'AssetExchangeFormCubit' and 'AssetExchangeFormBloc' ?

Do we want to unify them to one Bloc ?

File: lib/view/widgets/keypad.dart [hunk 6]
```diff
@@ -6,6 +6,28 @@ import 'package:simplio_app/view/widgets/back_gradient1.dart';
 import 'package:simplio_app/view/widgets/keypad_item.dart';
 import 'package:sio_glyphs/sio_icons.dart';
 
+// TODO - find a better name
+const int numpadBehaviorRange = 1000;
```
Comment (TomasPetrik) [inline lib/view/widgets/keypad.dart:10]: You can use something like 'numpadReservedActionsOffset' or 'reservedActionsStartIndex'

File: lib/logic/bloc/asset_exchange_form/asset_exchange_form_bloc.dart [hunk 7]
```diff
@@ -0,0 +1,234 @@
+import 'package:crypto_assets/crypto_assets.dart';
```
Comment (storkandstars) [inline lib/logic/bloc/asset_exchange_form/asset_exchange_form_bloc.dart:1]: remove this file for the PR

File: lib/logic/bloc/asset_exchange_form/asset_exchange_form_event.dart [hunk 8]
```diff
@@ -0,0 +1,60 @@
+part of 'asset_exchange_form_bloc.dart';
```
Comment (storkandstars) [inline lib/logic/bloc/asset_exchange_form/asset_exchange_form_event.dart:1]: remove this file for the PR

File: lib/logic/bloc/asset_exchange_form/asset_exchange_form_state.dart [hunk 9]
```diff
@@ -0,0 +1,191 @@
+part of 'asset_exchange_form_bloc.dart';
```
Comment (storkandstars) [inline lib/logic/bloc/asset_exchange_form/asset_exchange_form_state.dart:1]: remove this file for the PR

File: lib/view/routes/authenticated_router.dart [hunk 10]
```diff
@@ -464,9 +466,16 @@ class AuthenticatedRouter with PageBuilderMixin {
                     path: ':assetId/:networkId/exchange',
                     name: assetExchange,
                     pageBuilder: pageBuilder(
-                      builder: (state) => AssetExchangeScreen(
-                        sourceAssetId: state.params['assetId'],
-                        sourceNetworkId: state.params['networkId'],
+                      builder: (state) => BlocProvider(
```
Comment (storkandstars) [inline lib/view/routes/authenticated_router.dart:469]: remove the bloc provider for exchange bloc

File: lib/view/widgets/keypad.dart (hunk 6 above)
Comment (storkandstars) [inline lib/view/widgets/keypad.dart:10]: remove the enum implementation for this PR

File: lib/data/model/helpers/stringable_double.dart [hunk 11]
```diff
@@ -0,0 +1,163 @@
+import 'package:equatable/equatable.dart';
```
Comment (storkandstars) [inline lib/data/model/helpers/stringable_double.dart:1]: remove this type definition

File: lib/data/repositories/wallet_repository.dart [hunk 12]
```diff
@@ -63,7 +63,8 @@ class WalletRepository {
     String? name,
     String? mnemonic,
   }) async {
-    final m = mnemonic ?? sio.Mnemonic().generate;
+    final m = mnemonic ??
+        'inject father basket boss oyster kick cube enact novel tent liberty captain awesome title defense surround middle struggle clap milk notable carry exercise process';
```
Comment (storkandstars) [inline lib/data/repositories/wallet_repository.dart:67]: REMOVE THIS !!!!!

File: lib/data/http/services/swap_service.dart [hunk 13]
```diff
@@ -129,10 +129,12 @@ class SingleSwapBody {
 class CryptoAsset {
   final int assetId;
   final int networkId;
+  final double price;
```
Comment (storkandstars) [inline lib/data/http/services/swap_service.dart:132]: This change is not used yet. remove it

File: lib/data/http/services/buy_service.dart [hunk 14]
```diff
@@ -244,24 +277,98 @@ class FiatAsset {
...
+    return FiatAssetWithStringableDouble(
+      minimum: minimum,
+      maximum: maximum,
+      assetId: assetId,
+      amount: StringableDouble.string(amount.toString(), precision: 2),
+    );
+  }
+}
+
+class FiatAssetWithStringableDouble extends Equatable {
```
Comment (storkandstars) [inline lib/data/http/services/buy_service.dart:291]: Stringable double is oboslete

File: lib/view/routes/guards/protected_guard.dart [hunk 15]
```diff
@@ -178,10 +178,15 @@ class _Protection extends StatelessWidget {
                 key: const Key('protected-screen-numpad'),
                 displayEraseButton: true,
                 onTap: (value) async {
-                  context.read<PinVerifyFormCubit>().changeFormValue(value);
-                },
-                onErase: () {
-                  context.read<PinVerifyFormCubit>().eraseLastValue();
+                  if (value == NumpadValue.erase) {
```
Comment (storkandstars) [inline lib/view/routes/guards/protected_guard.dart:181]: remove implementation of NumpadValue enum

**Review by storkandstars — CHANGES_REQUESTED** [review]

**Pull request closed without merging**